### ISSUE TITLE
DELETE methods not allowed by apache journalist interface config

### ISSUE THREAD

**Issue by redshiftzero** [freedomofpress/securedrop#3977]
## Description

This is the issue causing https://github.com/freedomofpress/securedrop-client/issues/173

## Steps to Reproduce

1. Provision staging VMs
2. Get a valid API token [[ref]](https://gist.github.com/redshiftzero/934093c4f8e0f47c53e1add475345e34) 
3. Add a source via the source interface
4. Attempt to remove a star from a source (this is an idempotent operation so one should not need to first add the star to hit the endpoint):

```
curl -X DELETE -H "Content-Type: application/json" -H "Authorization: Token mytokenhere" --proxy socks5h://127.0.0.1:9150 mystagingserver.onion/api/v1/sources/my-source-uuid-here/remove_star
```

## Expected Behavior

`{"message":"Star removed"}`

## Actual Behavior

A 403 occurs, and we get redirected to the login page. 

This redirect-for-403 behavior only happens for the API in staging because Apache is [sending 403s to `/notfound`](https://github.com/freedomofpress/securedrop/blame/eaa2d45d371b64d5bb82317209d686c63063fec8/install_files/ansible-base/roles/app/templates/sites-available/journalist.conf#L42), causing the [`request.path`](https://github.com/freedomofpress/securedrop/blob/develop/securedrop/journalist_app/__init__.py#L136) to be `/notfound` (instead of something that begins with `thisismeonion.onion/api/blahhh`), meaning that code execution will continue until [the redirect here](https://github.com/freedomofpress/securedrop/blob/develop/securedrop/journalist_app/__init__.py#L140).

## Comments

The cause is the 403, and this is happening because of the [LimitExcept](https://httpd.apache.org/docs/current/mod/core.html#limitexcept) directive we are using in the Apache configs. These allow only GET POST and HEAD requests.

To prove this to oneself, you can edit `/etc/apache2/sites-enabled/journalist.conf` and add DELETE as an allowed request method.

This is similar to bugs #3772 and #3877. In light of this we should:

1. Revisit #1775 and,
2. Incorporate some API endpoint testing which would have detected all three bugs into at least the pre-release QA process (these could eventually be in CI as part of the external test client described in #3661)

**Comment by redshiftzero** [comment]
We discussed this bug today in the engineering meeting and decided that the way to resolve is to modify the journalist apache config, and request admins that want to use the journalist API to run `securedrop-admin install`. Otherwise we'll need to do another brittle apache config migration in the `postinst` of `securedrop-app-code` and given that the changes here are more extensive than other migrations we've made, we should avoid it if possible. 

This does mean that we'll need to be extremely careful when making any other Apache config changes in the future (until we have a better story regarding minimizing config drift across instances - insert obligatory ansible pull reference #3136 which is looking like a pretty good way forward in the medium term imho).
